documentation/reference/ci_cd.md

```diff
@@ -0,0 +1,489 @@
+
+**Inputs:**
+
+```typescript
+{
+  setup_images: string // default: '["upstream_db", "ingestion_server", "catalog", "api", "api_nginx"]'
+}
+```
+
+### `build-docs`
+
+Builds the documentation, including this Sphinx site, the frontend Storybook and
+the Tailwind config viewer and stores it at `/tmp/docs`. This compiled
+documentation is deployed to an appropriate location by the
+[`emit-docs`](#emit-docs) job.
+
+It needs the inputs `glotpress_username` and `glotpress_password` to download
+translations from GlotPress.
+
+```typescript
+{
+  glotpress_username: string // required, use secret `MAKE_USERNAME`
+  glotpress_password: string // required, use secret `MAKE_LOGIN_PASSWORD`
+}
+```
+
+## Helper jobs
+
+### `get-changes`
+
+Triggers the [`get-changes`](#get-changes) action to determine the changes. In
+addition to the `changes` from the action output, it emits some additional
+outputs that are used by subsequent jobs.
+
+**Outputs:**
+
+```typescript
+{
+  changes: string // JSON-encoded array of modified changesets
+  catalog: boolean // whether `changes` contains 'catalog'
+  ingestion_server: boolean // whether `changes` contains 'ingestion_server'
+  api: boolean // whether `changes` contains 'api'
+  frontend: boolean // whether `changes` contains 'frontend'
+  documentation: boolean // whether `changes` contains 'documentation'
+}
+```
+
+### `get-image-tag`
+
+Determines the tag to use for the Docker images. If the job is triggered via a
+workflow dispatch, the images are tagged with the `image_tag` input taken by the
+dispatch. If the job is triggered via a PR or a commit pushed to `main`, the
+images are tagged with the full 40-char hash
+([`github.sha`](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context))
+of the last commit.
+
+**Outputs:**
+
+```typescript
+{
+  image_tag: string // the tag to use for the Docker images
+}
+```
+
+### `determine-images`
+
+Determines which images to build and publish as a part of the workflow run. To
+speed up the workflow and avoid wasteful resource consumption, only a subset of
+all images are built by the workflow and a only subset of those images are
+published to GHCR. This job determines those images.
+
+| Change             | `upstream_db` | `catalog` | `ingestion_server` | `api` | `api_nginx` | `frontend` |
+| ------------------ | ------------- | --------- | ------------------ | ----- | ----------- | ---------- |
+| `catalog`          | 🛠️            | 🚀        |                    |       |             |            |
+| `ingestion_server` | 🛠             |           | 🚀                 | 🛠     |             |            |
+| `api`              | 🛠             |           | 🛠                  | 🚀    | 🚀          |            |
+| `frontend`         |               |           |                    |       |             | 🚀         |
+
+🚀 implies that the image is published to GHCR. 🛠️ implies that the image is
```

Comment:
This is excellent! I am noticing though that there are different characters used for the first hammer/wrench emoji and all the others 😮

Reply:
Sorry I do not understand, they are all 🛠 ("hammer and wrench").

Reply:
![image](https://user-images.githubusercontent.com/24264157/232655259-4a27fd99-0307-4191-94cf-3da1e81f230e.png)

I bet macOS isn't making a distinction between the two, but they are different:

```
>>> '🛠'.encode()
b'\xf0\x9f\x9b\xa0\xef\xb8\x8f'
>>> '🛠'.encode()
b'\xf0\x9f\x9b\xa0'
```

The first is the colourful version, the second is the plain black version. No idea what would have caused the difference though.

Reply:
Can you make them identical, I tried copying and pasting but Git shows no changes.

Reply:
I was able to do so in VSCode: https://github.com/WordPress/openverse/pull/1806

Reply:
Please feel free to merge the PR or push the commit to this branch directly!